atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/pool/CassandraService.java

```diff
@@ -367,6 +369,12 @@ public Optional<CassandraClientPoolingContainer> getRandomGoodHostForPredicate(
         return randomLivingHost.map(pools::get);
     }
 
```

Comment:
This might be a separate consideration, but for `com.palantir.atlasdb.keyvalue.cassandra.pool.CassandraService#getReachableProxies` should we be including the `inputHost` as one of the reachable proxies as that may be the client addressable hostname/IP? I was thinking something along the lines of:

```
    private Set<InetSocketAddress> getReachableProxies(String inputHost) throws UnknownHostException {
        int knownPort = getKnownPort();

        ImmutableSet.Builder<InetSocketAddress> proxies = ImmutableSet.builder();
        proxies.add(new InetSocketAddress(inputHost, knownPort));
        for (InetAddress address : InetAddress.getAllByName(inputHost)) {
            // It is okay to have reachable proxies that do not have a hostname
            proxies.add(new InetSocketAddress(address, knownPort));
        }
        return proxies.build();
    }
```

Reply:
That seems fine to me, though not super aware of the network-related implications